developers/contracts/cross-peg-out-endpoint-v2-01.md

```diff
@@ -0,0 +1,136 @@
+# cross-peg-out-endpoint-v2-01.clar
+
+- Location: `xlink/packages/contracts/bridge-stacks/contracts/cross-peg-out-endpoint-v2-01.clar`
+- [Deployed contract]()
+
+This technical document provides a detailed overview of the contract responsible for managing the peg-out process, enabling the transfer of `SIP-010` bridged tokens from the Stacks network to EVM-compatible blockchain networks as the EVM-based assets. The contract manages token transfers by validating amounts, applying fees, and maintaining a whitelist for authorized users if enabled. During the process, tokens are either burned or transferred (depending on the token's configuration) to a designated address on the EVM chain. The core functionalities of the contract are implemented through a series of public and governance functions, as described below.
+
+## Storage
+### `is-paused`
+| Data     | Type   |
+| -------- | ------ |
+| Variable | `bool` |
+
+A flag that indicates whether the peg-out process is active. If set to `true`, all peg-out operations are paused, preventing any new transactions. The contract is deployed in a paused state by default.
+
+### `use-whitelist`
+| Data     | Type   |
+| -------- | ------ |
+| Variable | `bool` |
+
+A flag that determines whether a whitelist is enforced for peg-out operations. If set to `true`, only addresses explicitly whitelisted can execute peg-out transactions. By default, this feature is disabled.
+
+### `whitelisted-users`
+| Data     | Type   |
+| -------- | ------ |
+| Variable | `principal` `bool` |
```

Comment:
I guess this should be a Map instead of a variable.. also check that I would add an arrow or something between `principal` and `bool`